CHANGELOG.md

```diff
@@ -26,6 +26,44 @@
 
     See [Migrating from generics to class attributes](https://django-components.github.io/django-components/latest/concepts/advanced/typing_and_validation/#migrating-from-generics-to-class-attributes) for more info.
 
+- The middleware `ComponentDependencyMiddleware` was removed as it is no longer needed.
+
+    The middleware served one purpose - to render the JS and CSS dependencies of components
+    when you rendered templates with `Template.render()` or `django.shortcuts.render()` and those templates contained `{% component %}` tags.
+
+    Now, the JS and CSS dependencies of components are automatically rendered,
+    even when you render Templates with `Template.render()` or `django.shortcuts.render()`.
+
+    - NOTE: If you rendered HTML with `Component.render()` or `Component.render_to_response()`, the JS and CSS were already rendered.
+
+    To disable this behavior, set the `DJC_DEPS_STRATEGY` context key to `"raw"`
+    when rendering the template:
+
+    ```py
+    # With `Template.render()`:
+    template = Template(template_str)
+    rendered = template.render(Context({"DJC_DEPS_STRATEGY": "raw"}))
```

Comment:
This feels a bit dirty to expose to users. A magical variable that you are passing. I wonder if it would make more sense if we provided our own version of Template that you can import if you want the change the deps_strategy? Feels a lot cleaner I think.

Reply:
Yeah, but unfortunately this is the best I could come up with.

The issue with having for example a custom `Template` class is that it might not work with other part of Django or even 3rd party packages that render templates.

Take for example just the `django.shortcuts.render()` method - you give it only the template name, and it creates an instance of `Template` for you. So you can't override that.

On the other hand, the context variables will be passed through no matter how the template is created and called.

Reply:
Is this something that we could change with our own template engine?

Reply:
Yeah, then we could have own reimplementations of e.g. `django.shortcuts.render()`, which could accept extra argument to set strategy.